daemon/start_windows.go

```diff
@@ -78,6 +79,14 @@ func (daemon *Daemon) getLibcontainerdCreateOptions(container *container.Contain
 			if err != nil {
 				continue
 			}
+
+			if data["GW_INFO"] != nil {
+				gwInfo := data["GW_INFO"].(map[string]interface{})
+				if gwInfo["hnsid"] != nil {
+					gwHNSID = gwInfo["hnsid"].(string)
```

Comment:
No. If the container has N endpoints which don't have gateway libnetwork only adds one gateway. This will be common across all endpoints. This is to provide external connectivity.